### PR TITLE
migrate to lefthook from husky

### PR DESCRIPTION
**Pull request by technophile-04**
### Description: 

Migrated to [lefhook](https://github.com/evilmartians/lefthook) from husky due #49. 

The main selling point of lefthook for us: 

1. It doesn't require `postinstall` to work
2. since its a single machine code binary also remove the need to mind external dependencies like (Husky + [lint-staged](https://github.com/okonet/lint-staged) , they both add roughly around fifften hundred dependencies to node_modules <- it was mentioned [here](https://arc.net/l/quote/rriulboi)

Fixes #48 

<details>

<summary>

### Notes

</summary>

### A bit about Git hooks

Git Hooks lets you fire some custom scripts when certain actions happens like (commit, push, etc). All you need is good knowledge of Bash scripts and internals of git. You can configure it in `./.git/hooks/pre-commit` . 

But while working on project you are interested in that project’s code—not the code that checks it. Hence tools like husky, lefthook were born. 

### Lefthook:

The important selling point is that : 

1. If you have repo with multiple languages like Ruby and TS you don’t need two separate tools like for TS [[husky](https://github.com/typicode/husky)](https://github.com/typicode/husky) and for ruby [[Overcommit](https://github.com/sds/overcommit)](https://github.com/sds/overcommit). 
2. Written in Go so fast and since its a single machine code binary also remove the need to mind external dependencies like (Husky + [[lint-staged](https://github.com/okonet/lint-staged)](https://github.com/okonet/lint-staged) , they both add roughly around *fifften hundred* dependencies to node_modules)

### Configuring Lefthook:

1. Add lefthook as dev dependency in `package.json`
    
    ```bash
    yarn add -D lefthook
    ```
    
2. Run `lefthook`
    
    ```bash
    lefthook install
    ```
    
    The above should create an `lefthook.yml` file
    
3. Example : 
    
    ```bash
    # lefthook.yml
    
    pre-commit:
      commands:
        eslint:
          glob: "*.{js,ts,jsx,tsx}"
          run: yarn eslint --fix {staged_files}
    ```
    
    Checkout this [[exmaple](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#run)](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#run) for various commands

</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,8 +39,7 @@
     "eslint": "^9.3.0",
     "eslint-config-prettier": "^9.1.0",
     "eslint-plugin-prettier": "^5.1.3",
-    "husky": "^9.0.11",
-    "lint-staged": "^15.2.4",
+    "lefthook": "^1.6.16",
     "pinst": "^3.0.0",
```
Comment (rin-st) [inline package.json:43]: should `pinst` be removed?

**Review by rin-st — APPROVED** [review]
It seems a great solution!

**Comment by carletex** [comment]
This is cool! Working great.

Should we consider `lefthook` for SE-2 too?

File: package.json (hunk 1 above)
Comment (technophile-04) [reply package.json]: Ohh shit missed it! removed at https://github.com/scaffold-eth/create-eth/pull/51/commits/23462beee6ee80ccec9ca840bfaf5f603806d872 Thanks !

**Comment by technophile-04** [comment]
> Should we consider lefthook for SE-2 too?

**Comment by Pabl0cks** [comment]
> Also cc @Pabl0cks could you please also give it a shot once on windows?

It's working nicely on Windows! Tried GH Desktop, Git bash and VSCode. Similar outputs:

image

**Pull request merged**